### PR TITLE
Render Courses with SIS Verification

### LINKED ISSUE #180: Display timetable courses on student's timetable inside course list

### PR DESCRIPTION
**Pull request by miaboloix**
Screen Shot 2021-04-18 at 7 34 52 PM
Screen Shot 2021-04-18 at 7 34 58 PM

### CODE REVIEW

**Review by yesterd3v — APPROVED** [review]
LGTM (but merge in new cs310, after James's PRs are all in, before merging this)

File: advising/urls.py
Comment (JJamesWWang) [inline advising/urls.py:40]: Looks like you added in an extra '/' here?

File: advising/urls.py
Comment (JJamesWWang) [inline advising/urls.py:43]: Did this actually work? 
I was thinking to swap the urls like this:
```
    # Get the courses for the student with this JHED and verify it against this student's timetable
    url(r'^advising/sis_courses/(?P<sem_name>.+)/(?P<year>[0-9]{4})/(?P<jhed>.+)/(?P<tt_name>.+)/$',
        advising.views.RegisteredCoursesView.as_view()),
    # Get the courses for the student with this JHED
    url(r'^advising/sis_courses/(?P<sem_name>.+)/(?P<year>[0-9]{4})/(?P<jhed>.+)/$',
        advising.views.RegisteredCoursesView.as_view()),
    # Get the courses for the requesting user
    url(r'^advising/sis_courses/(?P<sem_name>.+)/(?P<year>[0-9]{4})/$',
        advising.views.RegisteredCoursesView.as_view())
```
So that the url with 4 parameters is used first.

**Review by JJamesWWang — CHANGES_REQUESTED** [review]
* Need to remove extra '/' in /jhed/ url
* Should not have to interchange the position of jhed and tt_name in order for url endpoint to work (they are the same regex pattern)

The above two are the reasons the unit tests fail.
Try ordering the urls like I have mentioned in the above comment

File: advising/urls.py
Comment (miaboloix) [reply advising/urls.py:40]: that's the only way the two URLs work. If not, it always assumes that the URL without the tt_name is being used. So, if i ask for the view to print out the JHED - it gives me <timetable_name>/<jhed>. Adding the extra / means that if one the frontend, nothing is passed in for ttname, the URL it is calling is advising/sis_courses/<sem_name>/<year>//<jhed>/' since tt_name would be empty. It is also a solution to scrap that URL altogether and rely on the one below it - but this was my fix for now.

File: advising/urls.py
Comment (miaboloix) [reply advising/urls.py:43]: hmm in not sure if it mtaches in order like that but i can try

Comment (miaboloix) [reply advising/urls.py:43]: what is the use case for the last URL? Why don't we just use the first two?

Comment (JJamesWWang) [reply advising/urls.py]: That last URL was left in for backwards compatibility in case anyone was still using the endpoint without specifying the JHED. I can remove it (and the corresponding unit test) if it's no longer used.

**Review by JJamesWWang — APPROVED** [review]
LGTM!

**Pull request merged**
